gridy/Sources/Presentations/ProjectBoard/SettingsView.swift

```diff
@@ -6,82 +6,94 @@
+                                    .frame(width: 96, height: 30)
+                                    .blur(radius: 12)
+                            )
+                    }
                 }
-            }
-            .clipShape(Circle())
-            .onHover { isHovered in
-                profileHover = isHovered
-            }
-            .onTapGesture { profileClicked = true }
+                .clipShape(Circle())
+                .onHover { isHovered in
+                    viewStore.send(.hoveredItem(name: isHovered ? .profileEditButton : ""))
+                }
+        }
     }
 }
 
 extension SettingsView {
     var nameForm: some View {
-        HStack(alignment: .center, spacing: 0) {
-            Text("Name")
-                .font(.title3)
-                .fontWeight(.regular)
-                .foregroundStyle(Color.title)
-            Spacer()
-            RoundedRectangle(cornerRadius: 5)
-                .strokeBorder(textFieldHover ? Color.itemHovered : Color.item)
-                .foregroundStyle(textFieldHover ? Color.item : .clear)
-                .frame(width: 160, height: 24)
-                .overlay(alignment: .leading) {
-                    if textFieldSubmit {
-                        Text(textFieldText)
-                            .font(.title3)
-                            .fontWeight(.regular)
-                            .foregroundStyle(Color.title)
-                            .padding(.leading, 8)
-                    } else {
-                        TextField(textFieldText, text: $textFieldText)
+        WithViewStore(store, observe: { $0 }) { viewStore in
+            var textFieldSubmit: Binding<Bool> {
+                Binding(
+                    get: { viewStore.textFieldSubmit },
+                    set: { newValue in
+                        viewStore.send(.textFieldSubmit(bool: newValue))
+                    }
+                )
+            }
+            HStack(alignment: .center, spacing: 0) {
+                Text("Name")
+                    .font(.title3)
+                    .fontWeight(.regular)
+                    .foregroundStyle(Color.title)
+                Spacer()
+                RoundedRectangle(cornerRadius: 5)
+                    .strokeBorder(viewStore.hoveredItem == .profileTextFieldEditButton ? Color.itemHovered : Color.item)
+                    .foregroundStyle(viewStore.hoveredItem == .profileTextFieldEditButton ? Color.item : .clear)
+                    .frame(width: 160, height: 24)
+                    .overlay(alignment: .leading) {
+                        if viewStore.textFieldSubmit {
+                            Text(viewStore.profileName)
+                                .font(.title3)
+                                .fontWeight(.regular)
+                                .foregroundStyle(Color.title)
+                                .padding(.leading, 8)
+                        } else {
+                            TextField(
```

Comment:
```suggestiong

.onExitCommand {
                                viewStore.send(.popoverPresent(button: .settingButton, bool: false))
                                viewStore.send(.profileNameChanged(""))
                            }
                            
```

ESC를 눌렀다가 다시 세팅뷰를 popOver하면 이전에 썼던 text가 그대로 남아있어요.
이렇게 바꾸면 뷰도 사라지고 text도 "" 로 되겠어요!

Reply:
엥 suggestion이 왜 이렇게 들어갔지